server/rest/src/main/java/org/infinispan/rest/resources/ServerResource.java

```diff
@@ -89,34 +97,27 @@ public Invocations getInvocations() {
    private CompletionStage<RestResponse> listIgnored(RestRequest restRequest) {
       String cacheManagerName = restRequest.variables().get("cache-manager");
       DefaultCacheManager cacheManager = invocationHelper.getServer().getCacheManager(cacheManagerName);
-      NettyRestResponse.Builder builder = new NettyRestResponse.Builder();
 
-      if (cacheManager == null) return completedFuture(builder.status(NOT_FOUND).build());
+      if (cacheManager == null) return notFoundResponseFuture();
       CacheIgnoreManager ignoreManager = invocationHelper.getServer().getIgnoreManager(cacheManagerName);
       Set<String> ignored = ignoreManager.getIgnoredCaches();
-      try {
-         byte[] resultBytes = invocationHelper.getMapper().writeValueAsBytes(ignored);
-         builder.contentType(APPLICATION_JSON_TYPE).entity(resultBytes);
-      } catch (JsonProcessingException e) {
-         builder.status(HttpResponseStatus.INTERNAL_SERVER_ERROR).entity(e.getMessage());
-      }
-      return completedFuture(builder.build());
+      return asJsonResponseFuture(ignored, invocationHelper);
    }
 
    private CompletionStage<RestResponse> cacheManagers(RestRequest restRequest) {
-      return serializeObject(invocationHelper.getServer().cacheManagerNames());
+      return asJsonResponseFuture(invocationHelper.getServer().cacheManagerNames(), invocationHelper);
    }
 
    private CompletionStage<RestResponse> memory(RestRequest restRequest) {
-      return serializeObject(new JVMMemoryInfoInfo());
+      return asJsonResponseFuture(new JVMMemoryInfoInfo(), invocationHelper);
    }
 
    private CompletionStage<RestResponse> env(RestRequest restRequest) {
-      return serializeObject(ManagementFactory.getRuntimeMXBean().getSystemProperties());
+      return asJsonResponseFuture(System.getProperties(), invocationHelper);
```

Comment:
This needs to be wrapped in a SecurityAction

Reply:
Right, unfortunately this was never caught by the Rest tests (that enable security) since the ```ServerResource``` is only tested in the server module...

Reply:
https://issues.redhat.com/browse/ISPN-11979